sa4u_z3/main.py

```diff
@@ -340,12 +350,18 @@ def main():
         #    print(f'Ignored {_ignored} of {_num_exprs}')
         if not parsed_args.run_as_daemon:
             break;
-        time.sleep(10)
+        print(f'---END RUN---')
+        while not _run:
+            time.sleep(_time_to_sleep)
```

Comment:
We should use [sigwait](https://docs.python.org/3/library/signal.html#signal.sigwait) instead of sleeping.